contributors/design-proposals/sig-cli/get-describe-apiserver-extensions.md

```diff
@@ -0,0 +1,89 @@
+# Provide open-api extensions for kubectl get / kubectl describe columns
+
+Status: Pending
+
+## Motivation
+
+`kubectl get` and `kubectl describe` do not provide a rich experience
+for resources retrieved through federated apiservers and types not
+compiled into the kubectl binary.  Kubectl should support printing
+columns configured per-type without having the types compiled in.
+
+## Proposal
+
+Allow the apiserver to define the type specific columns that will be
+printed using the open-api swagger.json spec already fetched by kubectl.
+This provides a limited describe to only print out fields on the object
+and related events.
```

Comment:
how do you envision this interacting with https://github.com/kubernetes/kubernetes/pull/33900 ?